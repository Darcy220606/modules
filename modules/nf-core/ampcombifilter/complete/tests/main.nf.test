nextflow_process {

    name "Test Process AMPCOMBIFILTER_COMPLETE"
    script "../main.nf"
    process "AMPCOMBIFILTER_COMPLETE"

    tag "modules"
    tag "modules_nfcore"
    tag "ampcombifilter"
    tag "ampcombifilter/complete"

    test("ampcombifilter_complete - bacteroides_fragilis") {
        when {
            process {
                """
                input[0] =
                    [
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombifilter/sample_1_ampcombi.tsv', checkIfExists: true),
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombifilter/sample_2_ampcombi.tsv', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.tsv[0]).readLines()[0].contains("ampir"),
                    process.out.versions).match() }
            )
        }
    }

    test("ampcombifilter_complete - bacteroides_fragilis - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] =
                    [
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombifilter/sample_1_ampcombi.tsv', checkIfExists: true),
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/ampcombifilter/sample_2_ampcombi.tsv', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
