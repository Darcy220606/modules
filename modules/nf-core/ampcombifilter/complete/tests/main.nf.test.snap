{
    "ampcombifilter_complete - bacteroides_fragilis - stub": {
        "content": [
            {
                "0": [
                    "Ampcombi_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,2e1e89056562a255493bbc55e9f23db0"
                ],
                "log": [
                    
                ],
                "tsv": [
                    "Ampcombi_summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,2e1e89056562a255493bbc55e9f23db0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T09:06:43.011792746"
    },
    "ampcombifilter_complete - bacteroides_fragilis": {
        "content": [
            true,
            [
                "versions.yml:md5,2e1e89056562a255493bbc55e9f23db0"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T09:06:18.48198038"
    }
}